src/ipred.rs

```diff
@@ -181,56 +180,39 @@ unsafe fn splat_dc<BD: BitDepth>(
     dc: c_int,
     bd: BD,
 ) {
+    let stride = BD::pxstride(stride);
+    let width = width as usize;
     match BD::BPC {
         BPC::BPC8 => {
-            if !(dc <= 0xff as c_int) {
-                unreachable!();
-            }
+            assert!(dc <= 0xff);
             if width > 4 {
```

Comment:
@fbossen, is `width` a power of 2 here?  Otherwise, this check doesn't make sense, since `width = 5` would overwrite for an 8-byte `u64`.  If `width` is a power of 2, this makes sense, but `width >= 8` would be much clearer.